### PR TITLE
Windows ohai hints, fixes #433

### LINKED ISSUE #433: ohai_hints should be be created at c:\chef\ohai\hints when provisioning windows nodes
Looks like we should use ```ChefConfig::Config.platform_specific_path(ohai_hint)``` at  https://github.com/chef/chef-provisioning/blob/master/lib/chef/provisioning/convergence_strategy/precreate_chef_objects.rb#L129

Currently ```c:\etc\chef\ohai\hints\ec2.json``` gets created.

```ruby
      # Create the ohai file(s)
      def create_ohai_files(action_handler, machine)
        if convergence_options[:ohai_hints]
          convergence_options[:ohai_hints].each_pair do |hint, data|
            # The location of the ohai hint
            ohai_hint = "/etc/chef/ohai/hints/#{hint}.json"
            machine.write_file(action_handler, ohai_hint, data.to_json, :ensure_dir => true)
          end
        end
      end
```

Ohai looks for this file by checking platform_specific_path:

https://github.com/chef/ohai/blob/master/lib/ohai/config.rb#L35

```ruby
    def self.default_hints_path
      [ ChefConfig::Config.platform_specific_path('/etc/chef/ohai/hints') ]
    end
```

Which uses ```c:\chef\``` platform_specific_hints on windows:

https://github.com/chef/chef/blob/master/chef-config/lib/chef-config/config.rb#L50

```ruby
      if ChefConfig.windows?
        # turns \etc\chef\client.rb and \var\chef\client.rb into C:/chef/client.rb
        if env['SYSTEMDRIVE'] && path[0] == '\\' && path.split('\\')[2] == 'chef'
          path = PathHelper.join(env['SYSTEMDRIVE'], path.split('\\', 3)[2])
        end
      end
```

### PR DESCRIPTION
**Pull request by hh**
Base the write options on the target node

### CODE REVIEW

File: lib/chef/provisioning/convergence_strategy/precreate_chef_objects.rb [hunk 1]
```diff
@@ -127,6 +127,12 @@ def create_ohai_files(action_handler, machine)
           convergence_options[:ohai_hints].each_pair do |hint, data|
             # The location of the ohai hint
             ohai_hint = "/etc/chef/ohai/hints/#{hint}.json"
+            # It's in a different path on windows
+            if machine.machine_spec.data['normal']['chef_provisioning']['reference']['is_windows']
+              # we can't assume c: is the system drive, but it's pretty common
+              system_drive = "c:"
```
Comment (hh) [inline lib/chef/provisioning/convergence_strategy/precreate_chef_objects.rb]: Is ```machine.system_drive``` set at this point? If so we could use that.

File: lib/chef/provisioning/convergence_strategy/precreate_chef_objects.rb [hunk 2]
```diff
@@ -127,6 +127,12 @@ def create_ohai_files(action_handler, machine)
           convergence_options[:ohai_hints].each_pair do |hint, data|
             # The location of the ohai hint
             ohai_hint = "/etc/chef/ohai/hints/#{hint}.json"
+            # It's in a different path on windows
+            if machine.machine_spec.data['normal']['chef_provisioning']['reference']['is_windows']
+              # we can't assume c: is the system drive, but it's pretty common
+              system_drive = machine.execute_always('$env:SystemDrive').stdout.strip
```
Comment (hh) [inline lib/chef/provisioning/convergence_strategy/precreate_chef_objects.rb]: Trying to grab system_drive dynamically

**Comment by tyler-ball** [comment]
It looks like this also pulls in https://github.com/chef/chef-provisioning/commit/435d9b30232e3fc9529f36d9c965c7cc8f74e5d9 - @hh can you rebase this off master?

**Comment by hh** [comment]
@tyler-ball rebased, my bad

File: lib/chef/provisioning/convergence_strategy/precreate_chef_objects.rb [hunk 3]
```diff
@@ -127,6 +127,10 @@ def create_ohai_files(action_handler, machine)
           convergence_options[:ohai_hints].each_pair do |hint, data|
             # The location of the ohai hint
             ohai_hint = "/etc/chef/ohai/hints/#{hint}.json"
+            # It's in a different path on windows
+            if machine.machine_spec.data['normal']['chef_provisioning']['reference']['is_windows']
```
Comment (tyler-ball) [inline lib/chef/provisioning/convergence_strategy/precreate_chef_objects.rb]: Sorry, just thought of this too - you should be able to do `if machine.machine_spec.reference['is_windows']` - will you try that and see if it works?

**Pull request merged**